packages/viewer/src/lib/interactions/interactionApi.ts

```diff
@@ -204,7 +205,9 @@ export abstract class InteractionApi {
   public async beginInteraction(): Promise<void> {
     if (!this.isInteracting()) {
       this.interactionStartedEmitter.emit();
-      this.currentCamera = this.getScene().camera();
+      this.sceneLoadingPromise = this.getScene();
+      this.currentCamera = (await this.sceneLoadingPromise).camera();
```

Comment:
Could we await the line above instead? Like
```
this.sceneLoadingPromise = await this.getScene();
this.currentCamera = this.sceneLoadingPromise.camera();
```

Reply:
In this case we need the actual `Promise` so that the `endInteraction` method can `await` it in case the attempt comes in before the scene has been loaded, so unfortunately we need a setup like this. Open to other ideas on how to approach this though!